File: android-gif-drawable/src/androidTest/java/pl/droidsonroids/gif/ErrnoMessageTest.java
package pl.droidsonroids.gif;

import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;
import static org.junit.Assert.fail;

import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.TemporaryFolder;
import org.junit.runner.RunWith;

import java.io.File;

import androidx.test.ext.junit.runners.AndroidJUnit4;

@RunWith(AndroidJUnit4.class)
public class ErrnoMessageTest {

	@Rule
	public TemporaryFolder mTemporaryFolder = new TemporaryFolder();

	@Test
	public void errnoMessageAppendedToOpenFailed() throws Exception {
		final File nonExistentFile = new File(mTemporaryFolder.getRoot(), "non-existent");
		try {
			new GifDrawable(nonExistentFile);
			fail("Expected GifIOException to be thrown");
		} catch (GifIOException exception) {
			assertNotNull(exception.getMessage());
			assertTrue(exception.getMessage().contains("GifError 101: Failed to open given input: No such file or directory"));
		}
	}

	@Test
	public void errnoMessageAppendedToReadFailed() throws Exception {
		try {
			new GifDrawable(mTemporaryFolder.getRoot());
			fail("Expected GifIOException to be thrown");
		} catch (GifIOException exception) {
			assertNotNull(exception.getMessage());
			assertTrue(exception.getMessage().contains("GifError 102: Failed to read from given input: Is a directory"));
		}
	}

}

File: android-gif-drawable/src/androidTest/java/pl/droidsonroids/gif/ErrnoMessageTest.kt
package pl.droidsonroids.gif

import androidx.test.ext.junit.runners.AndroidJUnit4
import org.junit.Rule
import org.junit.Test
import org.junit.rules.ExpectedException
import org.junit.rules.TemporaryFolder
import org.junit.runner.RunWith
import pl.droidsonroids.gif.GifIOException
import java.io.File

@RunWith(AndroidJUnit4::class)
class ErrnoMessageTest {

    @get:Rule
    var mExpectedException = ExpectedException.none()

    @get:Rule
    var mTemporaryFolder = TemporaryFolder()

    @Test
    fun errnoMessageAppendedToOpenFailed() {
        mExpectedException.expect(GifIOException::class.java)

        mExpectedException.expectMessage("GifError 101: Failed to open given input: No such file or directory")

        val nonExistentFile = File(mTemporaryFolder.root, "non-existent")

        GifDrawable(nonExistentFile)
    }

    @Test
    fun errnoMessageAppendedToReadFailed() {
        mExpectedException.expect(GifIOException::class.java)

        mExpectedException.expectMessage("GifError 102: Failed to read from given input: Is a directory")

        GifDrawable(mTemporaryFolder.root)
    }
}